drivers/comparator/comparator_stm32_comp.c

```diff
@@ -0,0 +1,378 @@
+		.OutputPolarity = STM32_COMP_DT_INST_INV_OUT(inst),                                \
+		.OutputBlankingSource = STM32_COMP_DT_INST_BLANK_SEL(inst),                        \
+	}
+#else /* st_stm32h7_comp */
+#define STM32_COMP_DT_INST_COMP_CONFIG_INIT(inst)                                                  \
+	{                                                                                          \
+		.InputPlus = STM32_COMP_DT_INST_P_IN(inst),                                        \
+		.InputMinus = STM32_COMP_DT_INST_N_IN(inst),                                       \
+		.InputHysteresis = STM32_COMP_DT_INST_HYST_MODE(inst),                             \
+		.OutputPolarity = STM32_COMP_DT_INST_INV_OUT(inst),                                \
+		.OutputBlankingSource = STM32_COMP_DT_INST_BLANK_SEL(inst),                        \
+	}
+#endif /* st_stm32h7_comp */
+
+struct stm32_comp_config {
+	COMP_TypeDef *comp;
+	const struct stm32_pclken *pclken;
+	const struct pinctrl_dev_config *pincfg;
+	void (*irq_init)(void);
+	const uint32_t irq_nr;
+	const LL_COMP_InitTypeDef comp_config;
+	const uint32_t exti_line;
+	const uint8_t exti_line_number;
+	const bool lock_enable;
+	const bool miller_effect_hold_enable;
+};
+
+struct stm32_comp_data {
+	comparator_callback_t callback;
+	void *user_data;
+};
+
+__maybe_unused static bool stm32_comp_is_resumed(void)
+{
+#if CONFIG_PM_DEVICE
+	enum pm_device_state state;
+
+	(void)pm_device_state_get(DEVICE_DT_INST_GET(0), &state);
+	return state == PM_DEVICE_STATE_ACTIVE;
+#else
+	return true;
+#endif /* CONFIG_PM_DEVICE */
+}
+
+static int stm32_comp_get_output(const struct device *dev)
+{
+	const struct stm32_comp_config *cfg = dev->config;
+
+	return LL_COMP_ReadOutputLevel(cfg->comp);
+}
+
+static int stm32_comp_set_trigger(const struct device *dev, enum comparator_trigger trigger)
+{
+	const struct stm32_comp_config *cfg = dev->config;
+	struct stm32_comp_data *data = dev->data;
+	int ret = 0;
+
+	LL_EXTI_InitTypeDef exti_init_data = {0};
+
+	LL_EXTI_StructInit(&exti_init_data);
+	exti_init_data.LineCommand = ENABLE;
+	exti_init_data.Mode = LL_EXTI_MODE_IT;
+
+	if (cfg->exti_line_number < 32U) {
+		exti_init_data.Line_0_31 = cfg->exti_line;
+	} else if (cfg->exti_line_number < 64U) {
+		exti_init_data.Line_32_63 = cfg->exti_line;
+	} else {
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32h7_comp)
+		exti_init_data.Line_64_95 = cfg->exti_line;
+#endif /* st_stm32h7_comp */
+	}
+
+	switch (trigger) {
+	case COMPARATOR_TRIGGER_NONE:
+		exti_init_data.Trigger = LL_EXTI_TRIGGER_NONE;
+		break;
+
+	case COMPARATOR_TRIGGER_RISING_EDGE:
+		exti_init_data.Trigger = LL_EXTI_TRIGGER_RISING;
+		break;
+
+	case COMPARATOR_TRIGGER_FALLING_EDGE:
+		exti_init_data.Trigger = LL_EXTI_TRIGGER_FALLING;
+		break;
+
+	case COMPARATOR_TRIGGER_BOTH_EDGES:
+		exti_init_data.Trigger = LL_EXTI_TRIGGER_RISING_FALLING;
+		break;
+	}
+
+	irq_disable(cfg->irq_nr);
+	LL_COMP_Disable(cfg->comp);
+
+	ret = (int)LL_EXTI_Init(&exti_init_data);
```

Comment:
We need to get read of all usages of `LL_FOO_Init()` functions are they're not future proof (they won't be available in next STM32 series). Not the best time to introduce a new one.
Please rework the function so it is not used.

(Though, I guess this shouldn't be required following EXTI PR merge)